File: 0053. Maximum Subarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int csum = nums[0];
        int msum = nums[0];

        for (int i=1;i<nums.length;i++){
            if((csum+nums[i]>nums[i])){
                csum = csum+nums[i];
            }
            else{
                csum = nums[i];
            }

            if(csum>msum){
                msum = csum;
            }
        }

        return msum;
    }
}

File: 0053. Maximum Subarray/Solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        csum = nums[0]
        msum = nums[0]

        for i in range(1,len(nums)):
            csum = max(nums[i],csum+nums[i])
            msum = max(msum,csum)

        return msum 

        